bindings/azure/servicebusqueues/servicebusqueues.go

```diff
@@ -154,15 +154,24 @@ func (a *AzureServiceBusQueues) Invoke(ctx context.Context, req *bindings.Invoke
 		a.senderLock.Unlock()
 	}
 
+	a.logger.Infof("Invoke Binding incoming metadata: %+v", req.Metadata)
+
 	msg := &servicebus.Message{
-		Body: req.Data,
+		Body:                  req.Data,
+		ApplicationProperties: make(map[string]interface{}),
 	}
 	if val, ok := req.Metadata[id]; ok && val != "" {
 		msg.MessageID = &val
 	}
 	if val, ok := req.Metadata[correlationID]; ok && val != "" {
 		msg.CorrelationID = &val
 	}
+
+	// Include incoming metadata in the message to be used when it is read.
+	for k, v := range req.Metadata {
+		msg.ApplicationProperties[k] = v
```

Comment:
Shouldn't this contain a deny-list of keys we don't want included? For example `id` and `correlationID` (and more if needed) to avoid storing that data twice? Or perhaps it should be a sub-key, such as `req.Metadata["ApplicationProperties"]`?

Reply:
I think making it a subfield makes the incoming request too knowledgable of service bus queues. You're right we shouldn't store stuff twice though, so I'll add the deny-list there.